### ISSUE TITLE
Current working directory is different in gui mode.

### ISSUE THREAD

**Issue by drowzie** [HDLUtils/hdlregression#3]
In the vhdl testbench i want to open an external file. But when i run it in gui mode(-g arg), the current working directory seems to be from the where script is run.

When i run it without gui mode(-g arg) i need to go up directories as such: `constant c_proj_path : string := "../../../../";` whereas gui mode requires `constant c_proj_path : string := "";`

**Comment by drowzie** [comment]
Searched a bit through the code. I believe the difference is due to the path selected in the run command for GUI mode differs.
Gui run command is not using the path argument.
https://github.com/HDLUtils/hdlregression/blob/main/hdlregression/run/tcl_runner.py#L116
I guess it should be something like this?
```python
rc = self.runner.gui_run(command=command, path=test.get_test_path())
```

**Comment by HDLUtils** [comment]
Hi Christoffer,

Just to be clear for any other users wondering about the GUI and test folder structure:

- As a general rule a testcase should **not** be run in GUI mode. GUI mode is only for initial phases of setting up the test environment and debugging. Testcases should be run in terminal mode with a self checking testbench.
- The tests in HDLRegression is run in individual test folders, one folder for each testcase run. These test folders are moved and backed up on every regression run, i.e. a new empty test folder is created for each testcase on every run.
- Almost every run in GUI mode uses the wave view and users usually creates a wave.do file to save and load signals. If the GUI run is done inside the unique - and potentially moved - test folder, this wave.do file will also be moved and hard to load in a later GUI run without knowing the name of the backup folder etc.
This wave.do file is not moved by HDLRegression if the GUI run is done outside the HDLRegression test folder, thus the wave.do file can be loaded on the next GUI run if needed.

There are several ways to solve this challenge; add a load/save wave.do command to the GUI menu, change the path generics when run in GUI mode, or do the path change in the regression script when GUI mode is enabled.
Are you hard coding the project path (C_PROJ_PATH) inside your testbench? I recommend that you set this as a generic from the regression script.

Br.

**Comment by drowzie** [comment]
Yep, it is hardcoded as a constant and i guess the generic can do the same task to alter between gui mode and not.
